### PR TITLE
SDK generator bug fixes and response enhancements

### PR DESCRIPTION
**Pull request by TrevJohnson**
README.md 
- Reorganise information in README to separate ping-devops and local instance setup and exection

ApiResult returned instead of expected model type when generating operations for API classes ([Issue 10](https://github.com/Py-Ping/py-ping-fed-sdk/issues/10))
 - For APIs with multiple response codes the last response code dictates the type returned
 - In most cases the last return code to be processed is 422 which returns ApiResult
 - To avoid a generic return type for success status codes set the type before hand

Enum Types being override ([Issue 18](https://github.com/Py-Ping/py-ping-fed-sdk/issues/18))
- The get_enum_class function of the Process class uses a conditional statement to determine the corrent enum class
- The conditional statement uses the `in` operator rather than `==` which can lead to unexpected results when the string being tested is a sub-string of the actual model name
- Replace `in` with `==` to prevent this
- Add additional condition statements for missing enum types

Error responses for API Classes
- Return the json response for validation errors issued by Ping Federate to enable SDK users to troubleshoot configuration errors more easily
- Fix bug in API template where the string validation code is being checked for equality against an integer value of 422 which prevents those lines of the template being rendered 

Requests version
- Ping requests version to 2.31.0 to prevent conflicts caused when the generated SDK is used with other Python modules using different requests versions

Insecure requests warning
- Disable warnings issued when requests are made to Ping Federate instances using self signed certificates

Logging
- Added logging to provide more feedback during SDK generation

Envronment variables
- Simplified the environment variables that need to be set by the user executing against a local Ping Federate instance
- Created environment variables for admin credentials that can be set if required when using older versions of Ping Federate

### CODE REVIEW

File: src/helpers.py [hunk 1]
```diff
@@ -89,12 +89,21 @@ def get_py_type(json_type):
...
-    ping_user = os.environ.get(
-        "PING_IDENTITY_DEVOPS_ADMINISTRATOR", "administrator"
-    )
-    ping_pass = os.environ.get(
-        "PING_IDENTITY_DEVOPS_PASSWORD", "2FederateM0re"
-    )
+
+    if 'PING_IDENTITY_USER' in os.environ and 'PING_IDENTITY_SECRET' in os.environ:
+        ping_user = os.environ.get(
+            "PING_IDENTITY_USER", "Administrator"
```
Comment (galactose) [inline src/helpers.py]: heads up, if you're doing an explicit if statement checking if those environment variables are present you can remove the `Administrator` and `2Federate` references as get uses those as defaults if the variables aren't present https://www.tutorialspoint.com/python3/dictionary_get.htm

File: Pipfile [hunk 2]
```diff
@@ -23,6 +23,7 @@ sphinx = "*"
 guzzle-sphinx-theme = "*"
 tox = "*"
 json-delta = "*"
+flake8 = "*"
```
Comment (galactose) [inline Pipfile]: do you need flake8 linting in default packaging for a reason?

**Review by galactose — APPROVED** [review]

File: src/helpers.py (hunk 1 above)
Comment (TrevJohnson) [reply src/helpers.py]: Thanks for reviewing this @galactose  . I have removed the redundant defaults.

File: Pipfile (hunk 2 above)
Comment (TrevJohnson) [reply Pipfile]: Unnecessary flake8 package removed.

**Review by teddevaal — APPROVED** [review]

**Pull request merged**